indi-qhy/qhy_ccd.cpp

```diff
@@ -1937,45 +2043,50 @@ bool QHYCCD::ISNewNumber(const char *dev, const char *name, double values[], cha
         //////////////////////////////////////////////////////////////////////
         else if (!strcmp(name, ReadModeNP.name))
         {
-            uint32_t imageRMw, imageRMh;
-            double newReadMode = ReadModeN[0].value;
+            //NEW CODE - Fix read mode handling 
+            uint32_t currentReadMode;
 
-            double chipw, chiph, pixelw, pixelh;
             IUUpdateNumber(&ReadModeNP, values, names, n);
-            int rc;
-
-            // Current effective area for mode [NEW CODE]
-            uint32_t effROI_subX, effROI_subY, effROI_subW, effROI_subH;
-            rc = GetQHYCCDEffectiveArea(m_CameraHandle, &effROI_subX, &effROI_subY, &effROI_subW, &effROI_subH);
+            uint32_t newReadMode = ReadModeN[0].value;
+    
+            //Check if camera already is in the requested read mode
+            int rc = GetQHYCCDReadMode(m_CameraHandle, &currentReadMode);
             if (rc == QHYCCD_SUCCESS)
             {
-                LOGF_INFO("GetQHYCCDEffectiveArea: subX :%d subY: %d subW: %d subH: %d", effROI_subX, effROI_subY,
-                          effROI_subW, effROI_subH);
+                if (newReadMode == currentReadMode) {
+
+                    LOGF_INFO("Camera is already in read mode: %u", currentReadMode);
+                    return true;
+                }
             }
-            // Current effective area for mode [NEW CODE]
 
             // Set readout mode
-            rc = SetQHYCCDReadMode(m_CameraHandle, ReadModeN[0].value); // [NEW CODE] declaration int rc removed
+            rc = SetQHYCCDReadMode(m_CameraHandle, newReadMode); // [NEW CODE] declaration int rc removed
             if (rc == QHYCCD_SUCCESS)
             {
-                LOGF_INFO("Read mode updated to %.f", ReadModeN[0].value);
-                // Get resolution
-                GetQHYCCDReadModeResolution(m_CameraHandle, ReadModeN[0].value, &imageRMw, &imageRMh);
+                
+                currentReadMode = newReadMode;
+                
+                char currentReadModeName[255];
```

Comment:
char currentReadModeName[255] = {0};

Reply:
Yes, much better. It should be updated in the new commit.